hiplot_mlflow.py

```diff
@@ -0,0 +1,166 @@
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import uuid
+from typing import Union
+from urllib.parse import parse_qs, urlparse
+
+import hiplot
+from mlflow import search_runs, tracking
+from numpy import finfo, float64
+from pandas import DataFrame
+
+_positive_inf = finfo(float64).max
+_negative_inf = finfo(float64).min
+
+
+def fetch(name: str, include_tags: bool = False) -> hiplot.Experiment:
+    """Fetch experiments from the MLflow experiments tracking,
+    by experiment name.
+
+    Parameters
+    ----------
+    name: str
+        The name of the tracked experiment to look up
+    include_tags: bool
+        Whether to include tags as well in the resulting HiPlot experiment (False)
+
+    Returns
+    -------
+    hiplot.Experiment
+        The resulting experiment
+    """
+    return fetch_by_id(_get_experiment_id_from_name(name), include_tags)
+
+
+def fetch_by_id(id: Union[int, str], include_tags: bool = False) -> hiplot.Experiment:
+    """Fetch experiments from the MLflow experiments tracking,
+    by experiment ID.
+
+    Parameters
+    ----------
+    id: int|str
+        The ID if the experiment to look up
+    include_tags: bool
+        Whether to include tags as well in the resulting HiPlot experiment (False)
+
+    Returns
+    -------
+    hiplot.Experiment
+        The resulting experiment
+    """
+    df = search_runs([id])
+    return _create_experiment_from_dataframe(df, include_tags)
+
+
+def fetch_by_uri(uri: str) -> hiplot.Experiment:
+    """Fetch experiments from the MLflow experiments tracking,
+    by full URI.
+
+    The URI can take these forms:
+    * `mlflow://id/<id>` to look up experiments by ID
+    * `mlflow://name/<name>` to look up experiments by name
+
+    You can also add a query string to modify the results:
+    * Setting `tags` to any value will return results with
+      tags included. For example: `mlflow://id/<id>?tags=yes`
+
+    Parameters
+    ----------
+    uri: str
+        The fetcher URI to use
+
+    Returns
+    -------
+    hiplot.Experiment
+        The resulting experiment
+    """
+    # Only apply this fetcher if the URI matches our schema
+    parsed_uri = urlparse(uri)
+    if parsed_uri.scheme != "mlflow" or parsed_uri.netloc not in {"name", "id"}:
+        # It's not what we can handle
+        raise hiplot.ExperimentFetcherDoesntApply()
+
+    reference_type = parsed_uri.netloc
+    reference = parsed_uri.path.strip("/")
+    include_tags = parse_qs(parsed_uri.query).get("tags") is not None
+    if reference_type == "id":
+        return fetch_by_id(reference, include_tags)
+    else:
+        return fetch(reference, include_tags)
```

Comment:
Minor/subtle thing:

At the moment, this block and line 91 above are a bit coupled - this `else` statement assumes that the only way to reach this code is by having a `reference_type` of `"name"`. This is fine, but an alternative implementation is to remove that second check from line 91 and then do:

```suggestion
    if reference_type == "id":
        return fetch_by_id(reference, include_tags)
    elif reference_type == "name":
        return fetch(reference, include_tags)
    else:
        raise hiplot.ExperimentFetcherDoesntApply()
```

This then avoids this implicit coupling between two parts of the code and centralises all logic in a single place. With a small function like this the current implementation isn't really a problem, but I think this is a subtle improvement that aids the understandability and maintainability of the code/

Reply:
Extra point: would this `else` case actually be better raising an `ExperimentValidationError`?

Reply:
Agreed on the first part.
The error itself, yeah, it seems better, though I have second thoughts on using this `ExperimentValidationError` in the first place, it seems a bit internal to HiPlot instead of encouraged to used outside, by looking at their code, but it feels more fitting, so will use it for now.